public/js/navigator-js/integration/StateViewMap.js

```diff
@@ -77,20 +136,52 @@ this.navigatorjs.integration = this.navigatorjs.integration || {};
 			$container = $inside.length > 0 ? $inside.first() : $container;
 		}
 
-		var i = _orderedRecipes.indexOf(recipe) + 1,
-			length = _orderedRecipes.length,
-			testRecipe;
-		for (i; i < length; i++) {
-			testRecipe = _orderedRecipes[i];
+		var additionType = '' + (parentRecipe ? parentRecipe._type : 'BACKBONE') + ' > ' + recipe._type;
 
-			if (testRecipe.isInstantiated() && testRecipe.getViewInstance().$el.parent()[0] == $container[0]) {
-				testRecipe.getViewInstance().$el.before(recipe.getViewInstance().$el);
-				return;
-			}
-		}
+		switch (additionType) {
+			case 'BACKBONE > BACKBONE':
+				var i = _orderedRecipes.indexOf(recipe) + 1,
+				length = _orderedRecipes.length,
+				testRecipe;
+
+				for (i; i < length; i++) {
+					testRecipe = _orderedRecipes[i];
+
+					// If any other views have the same parent, add this element before
+					// those elements in the container element
 
-		// otherwise add on top
-		$container.append(recipe.getViewInstance().$el);
+					if (testRecipe.isInstantiated() && testRecipe.isMounted() && testRecipe.getRootEl().parent()[0] == $container[0]) {
+						testRecipe.getRootEl().before(recipe.getRootEl());
+						return;
+					}
+				}
+				$container.append(recipe.getRootEl());
+				break;
+
+			case 'BACKBONE > REACT':
+				var $proxy = $(document.createElement('div'));
+				$container.append($proxy);
+				ReactDOM.render(recipe._viewInstance, $proxy[0]);
+
+				break;
+
+			case 'REACT > REACT':
+				parentRecipe._showChild(recipe);
+
+				while (parentRecipe.getParentRecipe()) {
```

Comment:
Can you explain a little bit what you are doing here? 

What about when you have BACKBONE > REACT > REACT? Is that a valid recipe combination? It seems like this would set parentRecipe to be the BACKBONE recipe. In the BACKBONE > REACT case, you create a proxy to render the React component.